tests/network/vmi_istio.go

```diff
@@ -0,0 +1,304 @@
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				BeforeEach(func() {
+					peerAuthenticationRes := schema.GroupVersionResource{Group: "security.istio.io", Version: "v1beta1", Resource: "peerauthentications"}
+					peerAuthentication := generateStrictPeerAuthentication()
+					_, err = virtClient.DynamicClient().Resource(peerAuthenticationRes).Namespace(tests.NamespaceTestDefault).Create(context.Background(), peerAuthentication, metav1.CreateOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+				})
+				table.DescribeTable("request to VMI", func(port int) {
+					job, err := createJobCheckingVMIReachability(vmi, port)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					By("Waiting for the job to fail")
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+				},
+					table.Entry("client outside mesh should not reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+					table.Entry("client outside mesh should not reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+				)
+			})
+		})
+		Describe("Outbound traffic", func() {
+			var (
+				serverVMIAddress string
+				serverVMI        *v1.VirtualMachineInstance
+				testPort         = 4200
+			)
+
+			curlCommand := func(serverIP string, port int) string {
+				return fmt.Sprintf("curl -sD - -o /dev/null http://%s:%d | head -n 1\n", serverIP, port)
+			}
+
+			generateExpectedHTTPReturnCodeRegex := func(codeRegex string) string {
+				return fmt.Sprintf("HTTP\\/[123456789\\.]{1,3}\\s(%s)", codeRegex)
+			}
+
+			BeforeEach(func() {
+				serverVMI = libvmi.NewCirros(
+					libvmi.WithNetwork(v1.DefaultPodNetwork()),
+					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding([]v1.Port{}...)),
+				)
+				serverVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(serverVMI)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
+
+				By("Starting HTTP Server")
+				tests.StartHTTPServer(serverVMI, testPort)
+
+				By("Getting back the Server VMI IP")
+				serverVMI, err = virtClient.VirtualMachineInstance(serverVMI.Namespace).Get(serverVMI.Name, &metav1.GetOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				serverVMIAddress = serverVMI.Status.Interfaces[0].IP
+			})
+
+			// Istio envoy may intercept the request, we need to check the header response, not just the
+			// return code of curl, because if envoy proxy forbids a request, it responds with 502 Bad Gateway code
+			It("Should be able to reach kubevirt.io", func() {
+				err = console.SafeExpectBatch(vmi, []expect.Batcher{
```

Comment:
This is what I meant with the error comment; this `err` could be locally scoped. 

You don't need to do it, I'm just pointing it out.